File: 프로그래머스/lv2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = [] # Stack
    
    for num in number:
        # 스택이 비어있는 경우 일단 넣는다.
        if not answer:
            answer.append(num)
            continue
        # 지울 기회가 남아있다면,
        if k > 0:
            while answer[-1] < num:
                answer.pop()
                k -= 1
                if not answer or k <= 0:
                    break
        answer.append(num)
        
    answer = answer[:-k] if k > 0 else answer
    return ''.join(answer)

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
  const answer = [];
  for (let num of number) {
    while (k > 0 && answer[answer.length - 1] < num) {
      answer.pop(num);
      k -= 1;
    }
    answer.push(num);
  }
  return answer.join("").slice(0, number.length - k);
}